File: BackEnd/src/model/activity-model.ts
import { Activity } from "@prisma/client";
import { Decimal } from "@prisma/client/runtime/library";

export interface CreateActivityRequest {
  name: string;
  description: string;
  start_time: Date;
  end_time: Date;
  type: string;
  cost: number;
  day_id: number;
  location_id: number;
}


export interface ActivityUpdateRequest {
  name: string;
  description: string;
  day_id: number;
  start_time: Date;
  end_time: Date;
  type: string;
  cost: Decimal;
  location_id: number;
}

export interface ActivityResponse {
  id: number;
  name: string;
  description: string;
  start_time: Date;
  end_time: Date;
  type: string;
  cost: Decimal;
  location_id: number;
}

export function toActivityResponse(activity: Activity): ActivityResponse {
  return {
    id: activity.id,
    name: activity.name,
    description: activity.description,
    start_time: activity.start_time,
    end_time: activity.end_time,
    type: activity.type,
    cost: activity.cost,
    location_id: activity.location_id
  };
}

export function toActivityResponseList(prismaTodo: Activity[]): ActivityResponse[] {
  const result = prismaTodo.map((activity) => {
      return {
        id: activity.id,
        name: activity.name,
        description: activity.description,
        start_time: activity.start_time,
        end_time: activity.end_time,
        type: activity.type,
        cost: activity.cost,
        location_id: activity.location_id
      }
  })

  return result
}

File: FrontEndChris/app/src/main/java/com/example/alp_visualprogramming/models/ActivityModel.kt
package com.example.alp_visualprogramming.models

data class ActivityModel (
val id: Int,
val name: String,
val description: String,
val start_time:String,
val end_time: String,
val type: String,
val cost: Double,
val location_id: Int,
)

data class GetActivityResponse (
    val data: ActivityModel
)

data class GetAllActivityResponse (
    val data: List<ActivityModel>
)

data class ActivityRequest (
    val name: String,
    val description: String,
    val start_time:String,
    val end_time: String,
    val type: String,
    val cost: Double,
    val location_id: Int,
)